framework/source/class/qx/ui/form/core/VirtualDropDownList.js

```diff
@@ -484,6 +550,51 @@ qx.Class.define("qx.ui.form.core.VirtualDropDownList",
       var toBottom = viewPortHeight - distance.bottom;
 
       return toTop > toBottom ? toTop : toBottom;
+    },
+
+
+    /**
+     * Loop over all model items to recalculate the maximum list item width.
+     *
+     * @protected
+     */
+    _recalculateMaxListItemWidth : function () {
+      "use strict";
+      var maxWidth = 0;
+      var list = this.getChildControl("list");
+      var model = list.getModel();
+      if (model && model.length) {
+        model.forEach(function (item) {
+          var length;
+          if (typeof item === "string") {
+            // Approximately 7 pixels per character
+            length = item.length * 7;
```

Comment:
hmmm, that seems like a very bold assumption... I would rather like to see you measure the actual width of the string in the correct font.

Reply:
The problem is that because it is a virtual dropdown, the widgets get generated whenever they're needed; so I can't easily get the width. Is there a way to get the width of the string in a certain font?

Reply:
you should use `qx.bom.Label.getHtmlSize` for calculating the real width based on the styles of the content element. See also how it is done in https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/ui/basic/Label.js#L382